addons/interaction/functions/fnc_getGlassDoor.sqf

```diff
@@ -0,0 +1,67 @@
+/*
+ * Author: Phyma
+ * Find glass door.
+ *
+ * Arguments:
+ * 0: Distance <NUMBER>
+ * 1: House <OBJECT>
+ * 2: Door name <STRING>
+ *
+ * Return Value: 
+ * 0: Door Name <STRING>
+ *
+ * Example:
+ * [player, target] call ace_interaction_fnc_getGlassDoor
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params ["_distance", "_house", "_door"]; 
+
+private ["_animName", "_splitStr", "_newString", "_objDist", "_animate"];
+private _doorParts = [];
+private _doorPos = [];
+private _animate = animationNames _house;
+private _glassDoor = _door splitString "_"; 
+private _glassPos = (_house selectionPosition [(_glassDoor select 0) + "_" + (_glassDoor select 1) + "_effects", "Memory"]);     
+// Calculate all animation names so we know what is there   
+{           
+    _animName = _x;
+    if ((["door", _animName] call BIS_fnc_inString) && !(["locked", _animName] call BIS_fnc_inString) && !(["disabled", _animName] call BIS_fnc_inString) && !(["handle", _animName] call BIS_fnc_inString)) then {         
+        _splitStr = _animName splitString "_";             
+        _doorParts pushBack ((_splitStr select 0) + "_" + (_splitStr select 1) + "_trigger");
+    };
+} forEach _animate; 
+
+// Get the pos of all the door components and save the parts
+{
+    _doorPos pushBack (_house selectionPosition [_x, "Memory"]);
+} forEach _doorParts;
+
+// Calculate what door that is closest to the glass door
+private _lowestDistance = 0;    
+{       
+    private _objDist = _glassPos distance  _x;
+    
+    if (_objDist <= _distance) then {    
+        //Need to set the value in the beginning
+        if (_lowestDistance == 0) then {
+            _lowestDistance = _objDist;
+            _newString = (_doorParts select _forEachIndex) splitString "_";         
```

Comment:
Just use `private _splitStr =` here and in the if block below, as well as in L32 above.
They are unrelated and only temporary, so there is no need for them to be set to the same scope. Yet the concept is the same so they should use the same identifier.